### PR TITLE
Porting Sqlconnection.GetSchema (PR #19937)

### PR DESCRIPTION
**Pull request by saurabh500**
* Adding SqlConnection.GetSchema() support
Port of PR #19937 
Fixes https://github.com/dotnet/corefx/issues/19797

### CODE REVIEW

**Comment by saurabh500** [comment]
cc @karelz Pending Shiproom approval.

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlMetadataFactory.cs [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+                "on assemblies.assembly_id = types.assembly_id ";
+
+            // pre 9.0/Yukon servers do not have UDTs
+            if (0 > string.Compare(ServerVersion, _serverVersionNormalized90, StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+
+            // Execute the SELECT statement
+            SqlCommand command = connection.CreateCommand();
```
Comment (nil4) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlMetadataFactory.cs]: Shouldn't the use of `SqlCommand` be wrapped in a `using` block?

File: src/System.Data.SqlClient/src/System/Data/SqlClient/SqlMetadataFactory.cs [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+            // TODO: update this check once the server upgrades major version number!!!
+            // pre 9.0/Yukon servers do not have Table types
+            if (0 > string.Compare(ServerVersion, _serverVersionNormalized10, StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+
+            // Execute the SELECT statement
+            SqlCommand command = connection.CreateCommand();
```
Comment (nil4) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/SqlMetadataFactory.cs]: Same question as above: `using` block?

**Comment by karelz** [comment]
Marked as blocked until Shiproom approval. (@saurabh500 FYI I don't drive shiproom, so I assume you sent the email ...)

**Comment by saurabh500** [comment]
@karelz, yes I did

**Comment by karelz** [comment]
Great!

**Comment by saurabh500** [comment]
Shiproom approved. Waiting for CI to finish validations.

**Comment by karelz** [comment]
FYI: I think that shiproom wanted to hit the merge button into rel/2.0.0 - just heads up, please clarify with them,

**Comment by saurabh500** [comment]
@karelz Do you mean they wanted to merge this PR?

**Comment by karelz** [comment]
That's what I heard last time in the process - I am still confused about it myself, I didn't see any motivation for it - just ask them.

**Comment by saurabh500** [comment]
@karel, I have received an approval from Shiproom. I am waiting for CI to finish the build and I will merge this PR

**Pull request merged**